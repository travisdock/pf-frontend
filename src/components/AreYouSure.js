import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.onOpen}
            >
            <div className="table-popup">
                <p>{this.props.message}</p>
                <button className="delete" onClick={this.props.handleDelete} >Delete</button>
                <button className="close" onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;