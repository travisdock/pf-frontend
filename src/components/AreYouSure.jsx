import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.askIfSure}
            >
            <div className="table-popup">
                <p>Are You Sure?</p>
                <button onClick={this.props.handleDelete} >Delete</button>
                <button onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;