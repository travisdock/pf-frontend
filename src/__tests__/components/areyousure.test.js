import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AreYouSure from '../../components/AreYouSure';

const props = {
    open: "ask",
    closeModal: jest.fn(),
    onOpen: jest.fn(),
    message: "this is a message",
    handleDelete: jest.fn()
}

it('renders a <AreYouSure/> snapshot', () => {
    const component = renderer.create(<AreYouSure {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

describe('correct functions fire on button clicks', () => {
    test('handleDelete fires on button click', () => {
        const component = shallow(<AreYouSure {...props} />);
        component.find('button.delete').simulate('click')
        expect(props.handleDelete).toHaveBeenCalled();
    });
    test('closeModal fires on button click', () => {
        const component = shallow(<AreYouSure {...props} />);
        component.find('button.close').simulate('click')
        expect(props.closeModal).toHaveBeenCalled();
    });
});

