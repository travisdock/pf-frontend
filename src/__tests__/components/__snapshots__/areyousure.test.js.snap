// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a <AreYouSure/> snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "left": "0px",
        "position": "absolute",
        "top": "0px",
      }
    }
  />,
  <div
    className="popup-overlay"
    onClick={[Function]}
    style={
      Object {
        "background": "rgba(0, 0, 0,0.5)",
        "bottom": "0",
        "display": "flex",
        "left": "0",
        "position": "fixed",
        "right": "0",
        "top": "0",
        "zIndex": "999",
      }
    }
  >
    <div
      className="popup-content "
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(255, 255, 255)",
          "border": "1px solid rgb(187, 187, 187)",
          "margin": "auto",
          "padding": "5px",
          "position": "relative",
          "width": "50%",
        }
      }
    >
      <div
        className="table-popup"
      >
        <p>
          this is a message
        </p>
        <button
          className="delete"
          onClick={[MockFunction]}
        >
          Delete
        </button>
        <button
          className="close"
          onClick={[MockFunction]}
        >
          Close
        </button>
      </div>
    </div>
  </div>,
]
`;
